[Tan Yong Rui] iP

Fenway17 opened the pull request.

Commit:
Add Gradle support

Fenway17 committed:
"Level-1: added skeleton code"

Fenway17 committed:
Level-2: added Add and List commands

Fenway17 committed:
Level-3: added Mark As Done

Fenway17 committed:
Level-4: added todos, events, and deadlines classes. reorganized code.

Fenway17 committed:
A-TextUiTesting: Added semi-automated testing. Added part of DukeException.

Fenway17 committed:
Level-5: Finalized error handling for DukeException.

Fenway17 committed:
Level-6: added delete function, tidied up code, rearranged strings, added helper functions.

Fenway17 committed:
Added save function. Added load function. Restyled parts of code.

Fenway17 committed:
Level-8: Added date/time checking. Added date/time functions.

Fenway17 committed:
Merge branch 'branch-Level-7'

Fenway17 committed:
Merge branch 'branch-Level-7' into branch-Level-8

Fenway17 committed:
Merge branch 'branch-Level-8'

Fenway17 committed:
A-MoreOOP: Refactored code to multiple new classes. Added classes Ui, Storage, Parser, TaskList.

Fenway17 committed:
A-Packages: Repackaged a few java classes into task package.

Fenway17 committed:
Fixed imports due to repackaging.

Fenway17 committed:
Added junit. Added tests for Todo, Event, Deadline.

Fenway17 committed:
Packaged app as a jar file.

Fenway17 committed:
Added javadocs to most methods and classes.

Fenway17 committed:
Added javadocs to branch. (accidentally committed to wrong branch previously)

Fenway17 committed:
Merge branch 'branch-A-JavaDoc'

Fenway17 committed:
Altered and confirmed code complies with coding standard.

Fenway17 committed:
Merge branch 'branch-A-CodingStandard'

Fenway17 committed:
Added find function and related methods.

Fenway17 committed:
Merge branch 'branch-Level-9'

Fenway17 committed:
Update README.md

Fenway17 committed:
Merge branch 'add-gradle-support' of https://github.com/Fenway17/ip into branch-A-Gradle

Fenway17 committed:
Added gradle support. Added checkstyle.

Fenway17 committed:
Merge branch 'branch-A-Gradle'

NatalieTanML commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,79 @@
+    * Runs the Duke bot.
+    */
+    public void run() {
+        // print introduction message
```
Not sure if this comment is necessary as it's the same as the Javadoc header comment

NatalieTanML commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,79 @@
+        //         + "| | | | | | | |/ / _ \\\n"
+        //         + "| |_| | |_| |   <  __/\n"
+        //         + "|____/ \\__,_|_|\\_\\___|\n";
+        // System.out.println("Hello from\n" + logo);
```
If this code is not needed, might be good to remove it so its cleaner

NatalieTanML commented on src/main/java/chatbot/DukeException.java:
```diff
@@ -0,0 +1,24 @@
+package chatbot;
+/**
+* Duke class for the bot.
```
Perhaps `Custom exception class for the bot.` might be a more descriptive/fitting description

NatalieTanML commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,200 @@
+package chatbot;
+import java.time.LocalDate;
+// import java.time.format.DateTimeFormatter;
+// import java.time.temporal.ChronoUnit;
```
Unneeded code can be removed

NatalieTanML commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,200 @@
+    * @see      Storage
+    */
+    public static boolean parseText(
+            String inputText, TaskList taskList, Storage storage) { // returns true if bot should continue parsing text
```
Perhaps the comment can be moved out into the Javadoc instead? i.e. `@return true if bot should continue parsing text`

NatalieTanML commented on src/main/java/chatbot/Storage.java:
```diff
@@ -0,0 +1,168 @@
+    }
+
+    /**
+    * Saves tasks in taskList into the data file duke.txt at filePath
+    * <p>
+    * The fileDirectory denotes the folder containing duke.txt
```
Maybe putting these comments describing `filePath` and `fileDirectory` as Javadoc on lines 26-27 might be better

NatalieTanML commented on src/main/java/chatbot/Storage.java:
```diff
@@ -0,0 +1,168 @@
+    * The fileDirectory denotes the folder containing duke.txt
+    *
+    * @param  taskList      a TaskList containing all the tasks stored
+    * @throws DukeException throws a DukeException
```
I'd suggest listing down what would cause the DukeException instead, in this case it would be due to an unreadable file

NatalieTanML commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,83 @@
+    protected boolean isDone = false;
+    protected String taskName = "";
+
+    ////////////////////////////////////////////////////////////////
+    // Constructors
```
If you're using IntelliJ IDEA (not sure about other IDEs/text editors), there's a feature where you can mark out code regions using the following syntax:

```
//region Constructors

...

//endregion
```

It can allow code folding and overall looks cleaner as well.

NatalieTanML reviewed (commented):
Overall, the code is quite clean and easy to understand. I like that you've included some comments within the functions to describe what is happening at a particular step. Do remove the unused package imports/code. LGTM!

pyk595 commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,200 @@
+            String inputText, TaskList taskList, Storage storage) { // returns true if bot should continue parsing text
+        inputText = inputText.trim();
+        String[] inputStringArray = inputText.split(" ");
+        try {
```
Might be a bit cleaner if you create some specific exceptions and you handle all the exceptions in a single try-catch? Probably not arrowheading right now but might become a risk if you add on more functionality per case

pyk595 commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,200 @@
+    * @throws   DukeException throws a DukeException
+    * @see      Task
+    */
+    public static Task createNewTaskFromInput(String type, String inputText) throws DukeException {
```
I like how your method name is descriptive and yet not too long

pyk595 commented on src/main/java/chatbot/Storage.java:
```diff
@@ -0,0 +1,168 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+// import java.io.BufferedWriter;
```
Can remove unused imports

pyk595 commented on src/main/java/chatbot/Storage.java:
```diff
@@ -0,0 +1,168 @@
+                    dataEntry += deadline.getIsDone() ? "/;DONE" : "/;NOT_DONE";
+                    dataEntry += "/;" + deadline.getTaskName();
+                    dataEntry += "/;" + deadline.getDate();
+                    if (deadline.getTime() == null) {
```
A little bit of arrow heading here, you can consider using switch cases, then try encapsulating some of the other if-else statements into the various objects also

pyk595 commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,112 @@
+    *
+    * @return   The LocalDate date of the event
+    */
+    public LocalDate getDate() {
```
Good naming of methods as verbs!

pyk595 reviewed (commented):
Overall not much comments on code quality, good job!

DannyDakota commented on src/main/java/chatbot/TaskList.java:
```diff
@@ -0,0 +1,171 @@
+        String taskListLength = Integer.toString(this.taskList.size());
+        reply += "Very well Sir. I have added this task:";
+        reply += "\n   " + newTask.toString();
+        reply += "\n You now have " + taskListLength + " tasks in the agenda Sir.";
```
declaring your reply variable in such a manner is great for readability!

DannyDakota commented on src/main/java/chatbot/Storage.java:
```diff
@@ -0,0 +1,168 @@
+import java.io.PrintWriter;
+import java.time.LocalDate;
+// import java.time.format.DateTimeFormatter;
+// import java.time.temporal.ChronoUnit;
```
remove unused imports

DannyDakota commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,200 @@
+            String[] inputStringArray = inputText.split(" /by ");
+            try {
+                taskName = inputStringArray[0].substring(9);
+            } catch (StringIndexOutOfBoundsException exception) {
```
An alternative solution to this separating out the commands into their own individual classes that inherit from a Command Parent class, and in the classes subsequently parse the input to get the output required. E.g DeadlineCommand class and a runCommand() method to splice the string and return the required output. Would be geared toward a more object oriented design. Otherwise, this block of code was easy to understand too! :)

DannyDakota commented on src/main/java/chatbot/Storage.java:
```diff
@@ -0,0 +1,168 @@
+    public void loadData(TaskList taskList) throws DukeException {
+        try {
+            // File directory = new File("./");
+            // System.out.println(directory.getAbsolutePath()); // used to check directory java starts at
```
remove excess comments

DannyDakota commented on src/main/java/chatbot/Storage.java:
```diff
@@ -0,0 +1,168 @@
+                case "Todo": // to-do task
+                    newTask = new Todo(dataEntryArray[2], isDone);
+                    break;
+                case "Deadline": // deadline task
```
good use of switch cases here, code vv readable!! goodjob!

DannyDakota replied:
Overall, very clean code I don't have much to nitpick on! goodjob!!

Fenway17 committed:
Added GUI to the bot.

Fenway17 committed:
Created .jar file. Added some assertions.

Fenway17 committed:
taskList class: added assertions.

the list items will change as the user uses the bot and checking
the list size will ensure it is being changed properly

assertions on list sizes are added to ensure the list size changes
are correct

prevents regressions where list is not updated properly

Fenway17 committed:
Checked through for code quality.

Code quality ensures readability and helps with future changes.

Some minor changes have been done. Most code quality checks have
already been done in previous commits before committing changes.

Code quality should be ensured in every commit, not just a single
commit.

Fenway17 committed:
Merge pull request #1 from Fenway17/branch-A-Assertions

taskList class: added assertions.

Fenway17 committed:
Merge branch 'master' into branch-A-CodeQuality

Fenway17 committed:
Merge pull request #2 from Fenway17/branch-A-CodeQuality

Checked through for code quality.

Fenway17 committed:
Added friendlier syntax for commands. (C-FriendlierSyntax)

Commands require the full command name.

Added shorter ways to access the same commands via initials or
short form of commands.

Intuitive for users to use initials or common short forms of words
used for commands.

Fenway17 committed:
Added user guide and ui image.

Fenway17 committed:
Set theme jekyll-theme-merlot

Fenway17 committed:
Fixed some SLAP and docs checkstyles.
